pkg/controllers/dashboard/helm/repo_oci.go

```diff
@@ -95,6 +95,7 @@ func (o *OCIRepohandler) onClusterRepoChange(key string, clusterRepo *catalog.Cl
 	if err != nil {
 		return o.setErrorCondition(clusterRepo, err, originalStatus)
 	}
+	originalIndex := index
```

Comment:
I think this should be a deep copy

Reply:
Deepcopy is not available and writing a deepcopy function is too tricky. So I used json.marshal to compare.